File: LeetCode/ProblemSet/Volume (3400-3499)/LeetCode3499/LeetCode3499-Solution-01.go
package leetcode3499

import "math"

func maxActiveSectionsAfterTrade(s string) int {
	pre, cnt, maxEx, total := math.MinInt, 0, 0, 0
	for i, c := range s {
		cnt++
		if i == len(s) - 1 || s[i] != s[i + 1] {
			if c == '1' {
				total += cnt
			} else {
				maxEx = max(maxEx, pre + cnt)
				pre = cnt
			}
			cnt = 0
		}
	}

	return total + maxEx
}

File: LeetCode/ProblemSet/Volume (3400-3499)/LeetCode3499/LeetCode3499-Solution-01.py
from typing import *
from math import inf
from itertools import groupby

class Solution:
    def maxActiveSectionsAfterTrade(self, s: str) -> int:
        pre, maxEx, total = -inf, 0, 0
        for c, group in groupby(s):
            cnt = len(list(group)) 
            if c == '1':
                total = total + cnt
            else:
                maxEx = max(maxEx, pre + cnt)
                pre = cnt
        
        return total + maxEx